### PR TITLE
Partitioning for the Result table

### PR DESCRIPTION
**Pull request by ee-usgs**

### CODE REVIEW

File: src/main/java/gov/acwi/wqp/etl/CreateTableLike.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package gov.acwi.wqp.etl;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import java.util.HashMap;
+
+/**
+ * Creates a table 'like' another table.  It is an error if the table already exists.
+ */
+public class CreateTableLike extends SqlTemplateTasklet {
```
Comment (ee-usgs) [inline src/main/java/gov/acwi/wqp/etl/CreateTableLike.java:10]: New class needed b/c all the existing code only created tables with '_swap_' automatically shoved into the name.  Later, when the swap tables are 'installed' into service, '_swap_' is removed from the names.  The partition tables are constructed as children of a result_swap_[data source name] table.  The parent swap table has its name auto-edited, but the children do not and retain a unique name.  The unique partition table names are constructed like this:
result_[data source name]_YYYYMMDDHH_[from date]_to_[to date]
where _YYYYMMDDHH_ is replaced by the run time (down to the hour) of the ETL job.

File: src/main/java/gov/acwi/wqp/etl/partition/DateRangePartitionStrategy.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
+package gov.acwi.wqp.etl.partition;
```
Comment (ee-usgs) [inline src/main/java/gov/acwi/wqp/etl/partition/DateRangePartitionStrategy.java:1]: This class is the real logic of partitioning and decides what partitions to create based on configured parameters.  It is 'no fail', meaning that any possible set of configuration parameters will be adjusted or filled in to ensure that partitions are created, even if not ideal.  By leaving the end date and quarter date configuration null, partition tables should be created in a logical and (hopefully) optimal way going into the future.

File: src/main/java/gov/acwi/wqp/etl/result/table/CreateResultSwapTable.java [hunk 3]
```diff
@@ -1,24 +1,23 @@
 package gov.acwi.wqp.etl.result.table;
```
Comment (ee-usgs) [inline src/main/java/gov/acwi/wqp/etl/result/table/CreateResultSwapTable.java:1]: This existing tasklet had created only a non-partitioned result swap table named 'result_swap_[data source]'.  It has been modified to create a table of the same name, but now owning range partitions.

File: src/main/java/gov/acwi/wqp/etl/result/table/SetupResultTable.java [hunk 4]
```diff
@@ -1,21 +1,33 @@
 package gov.acwi.wqp.etl.result.table;
```
Comment (ee-usgs) [inline src/main/java/gov/acwi/wqp/etl/result/table/SetupResultTable.java:1]: This existing class had to be retrofitted.  Before it assumed that the tasks that needed to be run were static and known ahead of time.  Now we dynamically build the list of steps to be run based on the list of partition tables that need to be made.

File: README.md [hunk 5]
```diff
@@ -85,3 +116,9 @@ To additionally start up a Docker database and run the integration tests of the
 ```shell
 mvn verify -DTESTING_DATABASE_PORT=5437 -DTESTING_DATABASE_ADDRESS=localhost -DTESTING_DATABASE_NETWORK=wqpEtlCore
 ```
+
+To run integration tests in an IDE with debug support and without waiting for a docker start, start the DB in Docker via:
+```shell
+mvn docker:start
+```
+
```
Comment (orosz-usgs) [inline README.md:124]: Thanks for updating the README

Comment (ssoper-usgs) [reply README.md:124]: Doc update is great, yes.

**Review by ssoper-usgs — APPROVED** [review]

**Review by orosz-usgs — APPROVED** [review]
good job. times out on my laptop running the epa etl. We will just have to run in test...

**Pull request merged**